### PR TITLE
[ListItem] Improved ListItem Accordion

### LINKED ISSUE #3010: Require cycles
Require cycle: node_modules/react-native-elements/dist/list/ListItem.js -> node_modules/react-native-elements/dist/list/ListItemAccordion.js -> node_modules/react-native-elements/dist/list/ListItem.js

react-native-elements: v3.4.1

  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.4 => 0.63.4

### PR DESCRIPTION
**Pull request by arpitBhalla**
noAnimation was not working converted to animation : object | boolean

May Fixes #3010 as well

### CODE REVIEW

**Comment by pranshuchittora** [comment]
@arpitBhalla I understood the reason behind the proposed changes.
IMO the prop name is not flexible enough to incorporate other behaviours.
I would suggest a prop named `animation` which can be true or false.
In case of true, it can accept an object with animation configuration like easing etc.

**Comment by pranshuchittora** [comment]
Screenshot 2021-05-03 at 8 16 11 PM

❗ ❗ ❗  This PR is not merged how can this be deployed on the website. ❗ ❗ ❗

**Comment by arpitBhalla** [comment]
@pranshuchittora Sir it was not much useless prop already present, I just removed it

**Comment by pranshuchittora** [comment]
Great 👍🏼 
Can you merge the `animationDuration` into a new prop called `animation`

**Comment by arpitBhalla** [comment]
@pranshuchittora kindly review

File: src/list/ListItemAccordion.tsx [hunk 1]
```diff
@@ -10,10 +10,9 @@ export type ListItemAccordionProps = ListItemProps & {
   icon?: IconNode;
   expandIcon?: IconNode;
   content?: React.ReactNode;
-  noAnimation?: boolean;
   noRotation?: boolean;
   noIcon?: boolean;
-  animationDuration?: number;
```
Comment (pranshuchittora) [inline src/list/ListItemAccordion.tsx:16]: IMO `animation` should be an object | boolean , with various properties one of which can be `duration`.

Comment (arpitBhalla) [reply src/list/ListItemAccordion.tsx:16]: Sure Sir

File: src/list/ListItemAccordion.tsx [hunk 2]
```diff
@@ -10,10 +10,12 @@ export type ListItemAccordionProps = ListItemProps & {
...
+  animation?: {
+    type?: 'timing' | 'spring';
+    duration?: number;
+  };
```
Comment (arpitBhalla) [inline src/list/ListItemAccordion.tsx:18]: `useNativeDriver` can only be false cuz height is not supported by native drivers

Comment (pranshuchittora) [reply src/list/ListItemAccordion.tsx:18]: type of `animation` should be `boolean | {animation_object}`
Boolean can take only false which means the animation should be disabled.

Is there any other way around?

Comment (arpitBhalla) [reply src/list/ListItemAccordion.tsx:18]: It didn't look nice without animation I guess, will become a simple hide and show toggle

Comment (arpitBhalla) [reply src/list/ListItemAccordion.tsx]: Done

**Pull request merged**